src/java.base/share/classes/jdk/internal/loader/URLClassPath.java

```diff
@@ -439,34 +439,56 @@ private synchronized Loader getLoader(int index) {
                 continue;
             }
             // Otherwise, create a new Loader for the URL.
-            Loader loader;
+            Loader loader = null;
+            final URL[] loaderClassPathURLs;
             try {
                 loader = getLoader(url);
-                // If the loader defines a local class path then add the
-                // URLs as the next URLs to be opened.
-                URL[] urls = loader.getClassPath();
-                if (urls != null) {
-                    push(urls);
-                }
+                // If the loader defines a local class path then construct
+                // and add the URLs as the next URLs to be opened.
+                loaderClassPathURLs = loader.getClassPath();
             } catch (IOException e) {
-                // Silently ignore for now...
+                // log the error and close the unusable loader (if any)
+                if (DEBUG) {
+                    System.err.println("Failed to construct a loader or construct its" +
+                            " local classpath for " + url + ", cause:" + e);
```

Comment:
At some point we should probably get rid of the DEBUG stuff, it looks like it was left over from early JDK releases.

Reply:
The debug logs are currently enabled through the `sun.misc.URLClassPath.debug` system property. Do you mean we should completely get rid of the `sun.misc.URLClassPath.debug` system property in future?